core/src/main/java/org/apache/druid/indexer/partitions/DimensionRangePartitionsSpec.java

```diff
@@ -144,6 +146,17 @@ public List<String> getPartitionDimensions()
     return partitionDimensions;
   }
 
+  public DimensionRangeShardSpec createShardSpec(
```

Comment:
> Secretly choosing a partition dimension behind the scenes is probably not so great

This is just to be compatible with what single_dim like before. I do not make any new modification here.

Reply:
> It is better to have this logic in DeterminePartitionsJob since it is only used there. You can check the number of dimensions and create either a DimensionRangeShardSpec or a SingleDimensionShardSpec.

Done

Reply:
> This is just to be compatible with what single_dim like before my pr. I do not make any modification here.

Makes sense, we can make this fix in a later PR.